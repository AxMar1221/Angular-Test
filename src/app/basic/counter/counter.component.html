<app-navbar></app-navbar>
<div class="container mt-5">

  <h1>Conuter {{ counter }}</h1>
  <hr/>

  <button class="btn btn-danger mx-1" (click)="increaseBy(+1)">+1</button>
  <button class="btn btn-danger mx-1" (click)="increaseBy(-1)">-1</button>
</div>
