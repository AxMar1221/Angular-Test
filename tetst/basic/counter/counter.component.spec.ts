import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../../src/app/basic/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ CounterComponent ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hacer match con el snapshot', ()=>{
    expect( compiled ).toMatchSnapshot();
  })

  test('increaseBy debe de incrementar o decrementar basado en la opciÃ³n seleccionada', ()=>{
    component.increaseBy(5);
    expect( component.counter ).toBe(15);
  });

  test('hacer click en los botones de incrementar y decrementar en 1', ()=>{
    const buttons = compiled.querySelectorAll('button');
    buttons[0].click();
    expect( component.counter ).toBe(11);
    buttons[1].click();
    buttons[1].click();
    expect( component.counter ).toBe(9);
  });

  test(' al cambiar el counter debe de actualizar la etiqueta "h1"', ()=>{

    component.increaseBy(10);
    fixture.detectChanges();

    const h1 = compiled.querySelector('h1');
    expect( h1?.textContent).toContain('20');
  })
});
