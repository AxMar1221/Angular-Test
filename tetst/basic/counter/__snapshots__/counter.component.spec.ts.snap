// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent debe de hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <app-navbar />
  <div
    class="container mt-5"
  >
    <h1>
      Conuter 10
    </h1>
    <hr />
    <button
      class="btn btn-danger mx-1"
    >
      +1
    </button>
    <button
      class="btn btn-danger mx-1"
    >
      -1
    </button>
  </div>
</div>
`;
